src/components/Maps/MapPage.tsx

```diff
@@ -76,15 +82,34 @@ const MapPage = ({
 
   useEffect(() => {
     if (navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition((position: Position) => {
-        setCurrentLocationPin({
-          lat: position.coords.latitude,
-          lng: position.coords.longitude,
-        });
-      });
+      navigator.geolocation.getCurrentPosition(
+        (position: Position) => {
+          setCurrentLocationPin({
+            lat: position.coords.latitude,
+            lng: position.coords.longitude,
+          });
+        },
+        geolocationErrorCallback,
+        { enableHighAccuracy: true }
```

Comment:
We are trying this option for enhanced geolocation. Let me know if this causes issues.

Reply:
I don't notice much of difference. Latency was also not affected.